src/cli/key/sev/sign.rs

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::backend::sev::snp::launch::IdAuth;
+use crate::backend::ByteSized;
+
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::prelude::*;
+
+use anyhow::{bail, Context};
+use camino::Utf8PathBuf;
+use clap::Args;
+use p384::ecdsa::signature::Signer as _;
+use p384::ecdsa::SigningKey;
+use p384::elliptic_curve::sec1::Coordinates;
+use p384::pkcs8::DecodePrivateKey;
+use p384::EncodedPoint;
+
+/// Generate SEV Signature for provided SEV keys and write to file.
+#[derive(Args, Debug)]
+pub struct Options {
+    /// SEV P-384 private key in PEM form
+    #[clap(long, required = true)]
+    author_key: Utf8PathBuf,
+
+    /// SEV P-384 private key in PEM form
+    #[clap(long, required = true)]
+    id_key: Utf8PathBuf,
+
+    /// File path to write signature
+    #[clap(long, required = true)]
+    out: Utf8PathBuf,
+}
+
+pub fn sign_id_sev_key(
+    sev_author_key: &SigningKey,
+    sev_id_key: &SigningKey,
+) -> anyhow::Result<Vec<u8>> {
+    let mut id_auth = IdAuth {
+        id_key_algo: 1,   // ECDSA P-384 with SHA-384.
+        auth_key_algo: 1, // ECDSA P-384 with SHA-384.
+        ..Default::default()
+    };
+
+    let verifying_key: EncodedPoint = sev_id_key.verifying_key().to_encoded_point(false);
+    let (mut r, mut s) = match verifying_key.coordinates() {
+        Coordinates::Uncompressed { x, y } => (x.to_vec(), y.to_vec()),
+        _ => bail!("Invalid verifying key"),
+    };
+    // The r and s values have to be in little-endian order
+    r.reverse();
+    s.reverse();
+    // and are zero extended to the size of the components in the IdAuth struct.
+    id_auth.id_key.component.r[..r.len()].copy_from_slice(&r);
+    id_auth.id_key.component.s[..s.len()].copy_from_slice(&s);
+    // Sign the SEV signing key with the SEV author key. This will be removed in the future.
```

Comment:
Is this comment still relevant, or is it a leftover from when this function was copied over here?

Reply:
leftover .. thanks!